src/network/network_content.cpp

```diff
@@ -221,9 +221,10 @@ void ClientNetworkContentSocketHandler::RequestContentList(uint count, const Con
 		 * A packet begins with the packet size and a byte for the type.
 		 * Then this packet adds a uint16 for the count in this packet.
 		 * The rest of the packet can be used for the IDs. */
-		uint p_count = std::min<uint>(count, (COMPAT_MTU - sizeof(PacketSize) - sizeof(byte) - sizeof(uint16)) / sizeof(uint32));
+		uint p_count = std::min<uint>(std::min<uint>(UINT16_MAX, count),
```

Comment:
why this added `std::min` with `UINT16_MAX`?

Reply:
Hysterical development raisins. If packets were to become larger than ~256 KiB, then a large amount of requested IDs could overflow p_count. Since I started the development with the idea of really huge packets, it could overflow but I reduced the scope for this PR and leave the really breaking things to a potential future PR.